### ISSUE TITLE
Error with large number of binds, only in thin mode

### ISSUE THREAD

**Issue by Max-at-Prorank** [oracle/node-oracledb#1573]
1. What versions are you using?

```
process.platform: win32 (also tested separately in a linux environment)
process.version: v18.12.1
process.arch: x64
require('oracledb').versionString: 6.0.1
require('oracledb').oracleClientVersionString: 19.10.0.0.0
```

2. Is it an error or a hang or a crash?

Error

3. What error(s) or behavior you are seeing?

The issue only happens in thin mode. Thick mode works fine. The specific error varies, oddly enough. The most common is `NJS-103: unexpected message type 48 received` while some other ones are `NJS-103: unexpected message type 32 received` or `Cannot read properties of undefined (reading 'statement')`

```
Error: NJS-103: unexpected message type 48 received
    at Object.throwErr (D:\Code\node_modules\oracledb\lib\errors.js:591:10)
    at ExecuteMessage.processMessage (D:\Code\node_modules\oracledb\lib\thin\protocol\messages\base.js:218:14)
    at ExecuteMessage.processMessage (D:\Code\node_modules\oracledb\lib\thin\protocol\messages\withData.js:89:13)
    at ExecuteMessage.process (D:\Code\node_modules\oracledb\lib\thin\protocol\messages\base.js:197:12)
    at ExecuteMessage.process (D:\Code\node_modules\oracledb\lib\thin\protocol\messages\withData.js:510:11)
    at ExecuteMessage.decode (D:\Code\node_modules\oracledb\lib\thin\protocol\messages\base.js:188:10)
    at Protocol._decodeMessage (D:\Code\node_modules\oracledb\lib\thin\protocol\protocol.js:77:17)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Protocol._processMessage (D:\Code\node_modules\oracledb\lib\thin\protocol\protocol.js:148:9)
    at async ThinConnectionImpl.execute (D:\Code\node_modules\oracledb\lib\thin\connection.js:796:5)
    at async Connection.execute (D:\Code\node_modules\oracledb\lib\connection.js:860:16)
    at async Connection.<anonymous> (D:\Code\node_modules\oracledb\lib\util.js:162:14)
```

4. Include a runnable Node.js script that shows the problem.

```typescript
connection = await pool.getConnection();

const sql = `
      BEGIN
        ${Array(150).fill(null).map((_, i) => `OPEN :v_out_${i}_0 FOR select :a_${i}, :b_${i}, :c_${i} from dual;`).join('\n        ')}
      END;
    `;

const params = Array(150).fill(null).map((_, i) => ({ [`a_${i}`]: 1, [`b_${i}`]: 2, [`c_${i}`]: 3, [`v_out_${i}_0`]: { dir: oracledb.BIND_OUT, type: oracledb.CURSOR } }))
      .reduce((a, x) => ({ ...a, ...x }), {});

await connection.execute(sql, params);
```

If you reduce the "150" to, for instance, 5 or 10, there is no error. If you remove all the in bind variables, there is no error.

```typescript
connection = await pool.getConnection();

const sql = `
      BEGIN
        ${Array(5).fill(null).map((_, i) => `OPEN :v_out_${i}_0 FOR select :a_${i}, :b_${i}, :c_${i} from dual;`).join('\n        ')}
      END;
    `;

const params = Array(5).fill(null).map((_, i) => ({ [`a_${i}`]: 1, [`b_${i}`]: 2, [`c_${i}`]: 3, [`v_out_${i}_0`]: { dir: oracledb.BIND_OUT, type: oracledb.CURSOR } }))
      .reduce((a, x) => ({ ...a, ...x }), {});

await connection.execute(sql, params); // <---- No error
```

```typescript
connection = await pool.getConnection();

const sql = `
      BEGIN
        ${Array(150).fill(null).map((_, i) => `OPEN :v_out_${i}_0 FOR select 1 from dual;`).join('\n        ')}
      END;
    `;

const params = Array(150).fill(null).map((_, i) => ({ [`v_out_${i}_0`]: { dir: oracledb.BIND_OUT, type: oracledb.CURSOR } }))
      .reduce((a, x) => ({ ...a, ...x }), {});

await connection.execute(sql, params); // <---- No error
```

**Comment by anthony-tuininga** [comment]
I can replicate the issue and will get back to you on the source of the problem!

**Comment by Max-at-Prorank** [comment]
Great, thank you for looking into and the quick response.

**Comment by sharadraju** [comment]
@Max-at-Prorank The patch is available in `lib/thin/protocol/messages/withData.js` and will be incorporated in the 6.0.2 release.
Please replace the above mentioned file in your repository and check again.

**Comment by sharadraju** [comment]
This is now available as part of node-oracledb 6.0.2. See the [release announcement](https://github.com/oracle/node-oracledb/releases/tag/v6.0.2)
